queue/bytes_queue.go

```diff
@@ -199,32 +204,45 @@ func (q *BytesQueue) peekCheckErr(index int) error {
 		return errInvalidIndex
 	}
 
-	if index+headerEntrySize >= len(q.array) {
+	if index >= len(q.array) {
 		return errIndexOutOfBounds
 	}
 	return nil
 }
 
-func (q *BytesQueue) peek(index int) ([]byte, int, error) {
+func (q *BytesQueue) peek(index int) ([]byte, int, int, error) {
```

Comment:
Probably that's a good time to add comments regarding return params.

Reply:
@neal-zhu Can you fix this?

Reply:
Maybe the second returned value blockSize is kind of unnecessary? Since we can always get blockSize by ``len(data)``